### PR TITLE
:book: admission webhook for core types

### PR DESCRIPTION
**Pull request by mengqiy**
fixes: #1270

### CODE REVIEW

File: docs/book/src/reference/webhook-for-core-types.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+You need to ensure the path here match the path in the marker.
+
+## Deploy
+
+Deploying it is just like deploying a webhook server for CRD. You need to
+1) provision the serving certificate 2) deploy the server
+
+You can follow the [tutorial](https://book.kubebuilder.io/cronjob-tutorial/running.html).
```
Comment (DirectXMan12) [inline docs/book/src/reference/webhook-for-core-types.md]: minor nit: change this to a relative link to the markdown file (like `/cronjob-tutorial/running.md`).  The tool will change the link in the html.

File: docs/book/src/reference/webhook-for-core-types.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+	marshaledPod, err := json.Marshal(pod)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+}
+```
+
+If you need a client, just add the following InjectClient method for your
```
Comment (DirectXMan12) [inline docs/book/src/reference/webhook-for-core-types.md]: probably just suggest that they pass in the client by hand if they need it (at struct construction time).

**Comment by mengqiy** [comment]
Updated.
Also created https://github.com/kubernetes-sigs/controller-runtime/pull/785

File: docs/book/src/reference/webhook-for-core-types.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+**Note**: in order to have controller-gen generate the webhook configuration for
+you, you need to add markers. For example,
+`// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io`
+
+## Update main.go
+
+Now you need to register your handler in the webhook server.
+
+```go
+mgr.GetWebhookServer().Register("/mutate-v1-pod", &webhook.Admission{Handler: &podAnnotator{mgr.GetClient()}})
```
Comment (DirectXMan12) [inline docs/book/src/reference/webhook-for-core-types.md]: one last nit here -- field names + public

**Comment by mengqiy** [comment]
@DirectXMan12 Updated

File: docs/book/src/reference/webhook-for-core-types.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```go
+mgr.GetWebhookServer().Register("/mutate-v1-pod", &webhook.Admission{Handler: &podAnnotator{Client: mgr.GetClient()}})
+```
+
+You need to ensure the path here match the path in the marker.
+
+## Deploy
+
+Deploying it is just like deploying a webhook server for CRD. You need to
+1) provision the serving certificate 2) deploy the server
```
Comment (droot) [inline docs/book/src/reference/webhook-for-core-types.md]: Did you mean to list 1 & 2 as list ? They appear together in one line.

Comment (mengqiy) [reply docs/book/src/reference/webhook-for-core-types.md]: > Did you mean to list 1 & 2 as list ?

No. And it should not matter much :)

**Pull request merged**